core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1846,18 +1846,12 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     * 1. Claim Project from RBS
     * 2. Update Billing Account information on Google Project
     *
-    * @param billingProject
+    * @param billingAccount
     * @param span
+    * @param workspaceId
     * @return Future[(GoogleProjectId, GoogleProjectNumber)] of the project that we claimed from RBS
     */
-  private def setupGoogleProject(billingProject: RawlsBillingProject, span: Span = null, workspaceId: String): Future[(GoogleProjectId, GoogleProjectNumber)] = {
-    // We should never get here with a missing or invalid Billing Account, but we still need to get the value out of the
-    // Option, so we are being thorough
-    val billingAccount = billingProject.billingAccount match {
-      case Some(ba) if !billingProject.invalidBillingAccount => ba
-      case _ => throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"Billing Account is missing or invalid for Billing Project: ${billingProject}"))
-    }
-
+  private def setupGoogleProject(billingProject: RawlsBillingProject, billingAccount: RawlsBillingAccountName, span: Span = null, workspaceId: String): Future[(GoogleProjectId, GoogleProjectNumber)] = {
```

Comment:
I wasn't sure if I should pass in both the `billingProject` and the `billingAccount` here. It stank a bit of code smell, but we needed the billingAccount a level up, and didn't need the `projectPoolType` a level up. The other option would be to bump the match below up a level, so we would instead pass in a `RawlsBillingAccountName` and a `ProjectPoolType`